internal/state/store.go

```diff
@@ -407,60 +420,106 @@ func ABCIResponsesResultsHash(ar *tmstate.ABCIResponses) []byte {
 }
 
 // LoadABCIResponses loads the ABCIResponses for the given height from the
-// database. If not found, ErrNoABCIResponsesForHeight is returned.
-//
-// This is useful for recovering from crashes where we called app.Commit and
-// before we called s.Save(). It can also be used to produce Merkle proofs of
-// the result of txs.
+// database. If the node has DiscardABCIResponses set to true, ErrABCIResponsesNotPersisted
+// is persisted. If not found, ErrNoABCIResponsesForHeight is returned.
 func (store dbStore) LoadABCIResponses(height int64) (*tmstate.ABCIResponses, error) {
+	if store.discardABCIResponses {
+		return nil, ErrABCIResponsesNotPersisted
+	}
+
 	buf, err := store.db.Get(abciResponsesKey(height))
 	if err != nil {
 		return nil, err
 	}
-	if len(buf) == 0 {
 
+	if len(buf) == 0 {
 		return nil, ErrNoABCIResponsesForHeight{height}
 	}
-
 	abciResponses := new(tmstate.ABCIResponses)
 	err = abciResponses.Unmarshal(buf)
 	if err != nil {
 		// DATA HAS BEEN CORRUPTED OR THE SPEC HAS CHANGED
 		tmos.Exit(fmt.Sprintf(`LoadABCIResponses: Data has been corrupted or its spec has
-                changed: %v\n`, err))
+				changed: %v\n`, err))
 	}
 	// TODO: ensure that buf is completely read.
 
 	return abciResponses, nil
 }
 
+// LoadLastABCIResponses loads the last ABCI response specifically after a crash
+// We conduct a sanity check here which compares the height from the last ABCIResponse to the
+// height that is given
+//
+// This is used for recovering from crashes where we called app.Commit and
+// before we called s.Save()
```

Comment:
Can we define latest a little more specifically? Maybe something like this? 

```suggestion
// LoadLastABCIResponses loads the ABCIResponses from the most recent height. 
// The height parameter is used to ensure that the response corresponds to the latest height.
// If not, an error is returned. 
//
// This method is used for recovering in the case that we called the Commit ABCI
// method on the application but crashed before persisting the results. 
```

I would also make sure to end comment sentences with periods. 

Edit: I see that this was borrowed from below, but I think we might as well clean it up a little while we're here.